common/springOverrides.lua

```diff
@@ -41,3 +41,29 @@ if Spring.GetModOptions then
 		return table.copy(modOptions)
 	end
 end
+
+if Spring.Echo then
+	local echo = Spring.Echo
+	local printOptions = { pretty = true }
+
+	local function multiEcho(...)
+		local args = table.pack(...)
+		local tableIndexes = {}
+
+		for index = 1, args.n do
+			local value = args[index]
+
+			if type(value) == 'table' then
+				table.insert(tableIndexes, index)
+			end
+		end
+
+		echo(unpack(args, 1, args.n))
```

Comment:
When I try printing out just a single table, I get output like this:

```
TABLE: 1, 2, 3
{
  1,
  2,
  3,
}
```

Apparently (I didn't know about this before) the engine [attempts to print out single tables as if they are arrays](https://github.com/beyond-all-reason/spring/blob/0787116093846d5e49f62fe3f83066e695cefcd0/rts/Lua/LuaUtils.cpp#L1253).

Given that the engine behavior is redundant and also lacking the features we already have in even the old `TableEcho`, I think we could do something like this (just some quick code to demonstrate what I'm thinking):

```suggestion
		local echoArgs = {}
		for index = 1, args.n do
			local value = args[index]

			if type(value) == 'table' then
				table.insert(tableIndexes, index)
				table.insert(echoArgs, "<table>")
			else
				table.insert(echoArgs, value)
			end
		end

		echo(unpack(echoArgs))
```